getsolus/solbuild
# Initial layer caching implementation

WIP.

Current issues:
- [x] If you Ctrl+C or something catastrophic happens during layer creation (i.e. the fetching of dependencies), the incomplete layer is not deleted. It should be deleted. When this happens, find the hash at the start of the output of `solbuild build` and `sudo rm -rf /var/cache/solbuild/layers/<hash>`. Or just `sudo rm -rf /var/cache/solbuild/layers` to delete all layers.
- [x] It seems like 32bit dependencies are not installed correctly.
- [ ] Display the hash of the layer not only at the start of the build, but also at the end of the build as well.

This PR shouldn't touch anything outside of `/var/cache/solbuild/layers`, so if anything catastrophic happens, just run `sudo solbuild dc` and start over.

Test instructions:
1. Ensure you have the `xxhash` package installed because you need the `xxxh128sum` binary in your PATH.
2. `gh pr checkout 95` or checkout this branch however you prefer to.
3. `make`
4. Now you should have a `solbuild` binary in `./bin/solbuild`. Use this `solbuild` as how you would normally build packages. You should notice that after you build a package once, when you build it the second time, it should jump straight to the setup phase, skipping the step of installing dependencies.
